java-engine/src/main/java/io/getunleash/engine/UnleashEngine.java

```diff
@@ -95,8 +95,8 @@ private static int[] buildCustomStrategyResults(
         continue;
       }
       int keyOffset = builder.createString(entry.getKey());
-      int valueOffset = builder.createString(entry.getValue().toString());
-      int propOffset = PropertyEntry.createPropertyEntry(builder, keyOffset, valueOffset);
+      int propOffset =
+          PropertyEntry.createPropertyEntry(builder, keyOffset, entry.getValue() ? 1 : 0);
```

Comment:
Embarrassing bug fix that totally didn't take me 20 minutes to find. If you send a String as a boolean then the first bit tends to be set and the Rust side tends to see that as a `true`. So instead you should just pass booleans as booleans, which in practice, is a single bit